os/include/tinyara/mm/mm.h

```diff
@@ -302,21 +302,21 @@ struct mm_allocnode_s {
 /* This describes a free chunk */
 
 struct mm_freenode_s {
-	mmsize_t size;				/* Size of this chunk */
 	mmsize_t preceding;			/* Size of the preceding chunk */
+#ifdef CONFIG_DEBUG_MM_HEAPINFO
```

Comment:
1. Could you let me know why this is necessary for free node?
2. When we free some node, the mm merges neighbor free nodes. Then this could give unwanted information.

Reply:
If a user calls mm_free, and then continues to use the same node, he might cause heap corruption. In such case, this might provide us the information about the process which had allocated the node. This scenario might not be true in most cases, but in some rare case, it will help us for debugging.